app/controllers/physical_server_controller.rb

```diff
@@ -36,4 +36,68 @@ def textual_group_list
     []
   end
   helper_method :textual_group_list
+
+  # Handles buttons pressed on the toolbar
+  def button
+    # Get the list of servers to apply the action to
+    servers = retrieve_servers
+
+    # Apply the action (depending on the button pressed) to the servers
+    apply_action_to_servers(servers)
+  end
+
+  private
+
+  # Maps button actions to actual method names to be called and the
+  # corresponding result status messages to be displayed
+  ACTIONS = {"physical_server_power_on"         => [:power_on,         _("Power On")],
+             "physical_server_power_off"        => [:power_off,        _("Power Off")],
+             "physical_server_restart"          => [:restart,          _("Restart")],
+             "physical_server_blink_loc_led"    => [:blink_loc_led,    _("Blink LED")],
+             "physical_server_turn_on_loc_led"  => [:turn_on_loc_led,  _("Turn On LED")],
+             "physical_server_turn_off_loc_led" => [:turn_off_loc_led, _("Turn Off LED")]}.freeze
+
+  # Returns a list of servers to which the button action will be applied
+  def retrieve_servers
```

Comment:
Don't add this method, please.

Instead call `find_records_with_rbac(PhysicalServer)`

that way you get proper RBAC checking on the items and you don't repeat yourself.